### DISCUSSION TITLE
Is it possible to use Vegafusion with DuckDB Wasm

### DISCUSSION THREAD

**Discussion by michaelzwong** [vega/vegafusion#467] [Q&A]
Hello team,

Thanks for all the great work put into this repository! In my current React project, I am using react-vega and plan to migrate to a framework that can handle more rows of data. I am already running a DuckDB WASM instance and wonder if VegaFusion has a suggested solution to connect to it (#394)

**Comment by jonmmease** [comment]
Hi @michaelzwong, thanks for the kind words!

I think it would be possible to compile VegaFusion's runtime to WASM and integrate with DuckDB, but I haven't looked into it beyond what I wrote up in #394.  I'd be interested to hear more about your use case though. Like how large the datasets are that you're working with, and whether you need full interactivity during rendering.

**Reply by michaelzwong** [reply]
Hi Jon,

I am working with datasets that have 1MM+ rows/data points and need full interactivity when the charts are rendering. Ideally, I'd be able to integrate Vega with external react components

**Reply by cboettig** [reply]
:heavy_plus_sign:   This would be amazing to have fully in WASM.  Using altair/vega, duckdb, and streamlit has let us prototype examples on this scale, but a fully client-side solution would be much better!

**Reply by jonmmease** [reply]
Good news, VegaFusion 2.0 will support full compilation to wasm. The first version 2 release candidate was released over the weekend. Here are the draft docs for the updated version of vegafusion-embed: https://vegafusion.io/v2/features/embed.html. A demo web app is at https://github.com/vega/vegafusion/tree/main/examples/editor-demo.

Note that this isn't directly integrated with Altair in any way. I'd be interested to hear more about the workflow you'd like to see here.
